hypothesis-python/tests/conftest.py

```diff
@@ -81,10 +82,13 @@ def sleep(naptime):
     def freeze():
         frozen[0] = True
 
-    monkeypatch.setattr(time_module, "time", time)
-    monkeypatch.setattr(time_module, "monotonic", time)
-    monkeypatch.setattr(time_module, "perf_counter", time)
-    monkeypatch.setattr(time_module, "sleep", sleep)
+    def _patch(name, fn):
+        monkeypatch.setattr(time_module, name, wraps(getattr(time_module, name))(fn))
```

Comment:
I may be misreading, but do we need an extra layer of wrapping here, since `wraps` modifies the underlying function?
```
wraps(...)(copy(fn))
```
where copy could be an inlined lambda

Reply:
In principle I think yes, but in practice a single patch seems fine (https://github.com/HypothesisWorks/hypothesis/issues/3642#issuecomment-1670931815) and so I don't think it's worth using a slightly uglier and slower form.